app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt

```diff
@@ -73,69 +79,49 @@ class InstallerViewModel(
     private var inputFile: File? = null
 
     private var installedApp: InstalledApp? = null
-    var isInstalling by mutableStateOf(false)
+    internal var isInstalling by mutableStateOf(false)
         private set
-    var installedPackageName by mutableStateOf<String?>(null)
+    internal var installedPackageName by mutableStateOf<String?>(null)
         private set
-    val appButtonText by derivedStateOf { if (installedPackageName == null) R.string.install_app else R.string.open_app }
+    internal val appButtonText by derivedStateOf { if (installedPackageName == null) R.string.install_app else R.string.open_app }
 
     private val workManager = WorkManager.getInstance(app)
 
-    private val _progress: MutableStateFlow<ImmutableList<Step>>
-    private val patcherWorkerId: UUID
-    private val logger = ManagerLogger()
-
-    init {
-        // TODO: navigate away when system-initiated process death is detected because it is not possible to recover from it.
-
-        viewModelScope.launch {
-            installedApp = installedAppRepository.get(packageName)
-        }
-
-        val (selectedApp, patches, options) = input
-
-        _progress = MutableStateFlow(
-            PatcherProgressManager.generateSteps(
-                app,
-                patches.flatMap { (_, selected) -> selected },
-                selectedApp
-            ).toImmutableList()
-        )
-
-        patcherWorkerId =
-            workerRepository.launchExpedited<PatcherWorker, PatcherWorker.Args>(
-                "patching", PatcherWorker.Args(
-                    selectedApp,
-                    outputFile.path,
-                    patches,
-                    options,
-                    _progress,
-                    logger,
-                    setInputFile = { inputFile = it }
-                )
+    private val _progress: MutableStateFlow<ImmutableList<Step>> = MutableStateFlow(
+        PatcherProgressManager.generateSteps(
+            app,
+            input.selectedPatches.flatMap { (_, selected) -> selected },
+            input.selectedApp
+        ).toImmutableList()
+    )
+
+    internal val progress = _progress.asStateFlow()
```

Comment:
```suggestion
    val progress = _progress.asStateFlow()
```
Is internal necessary?